.github/workflows/build_and_test.yml

```diff
@@ -10,18 +10,31 @@ jobs:
   unit_tests:
     strategy:
       matrix:
-        os: [ windows-latest, macos-11, macos-12, ubuntu-20.04, ubuntu-22.04 ]
+        os: [ windows-latest, macos-12, macos-latest, ubuntu-20.04, ubuntu-latest ]
     runs-on: ${{ matrix.os }}
     steps:
-      - uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8  # v2
-      - name: Set up Python 3.8
-        uses: actions/setup-python@d27e3f3d7c64b4bbf8e4abfb9b63b83e846e0435  # v4.5.0
+
+      - uses: actions/checkout@v4
+
+      - name: Set up Python 3.9
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.9
```

Comment:
I wonder if we should just go further and push to 3.10 or 3.11 or 3.12 at this point.

Reply:
I agree and will do this for the next group of revisions.